src/main/java/picard/sam/SamErrorMetric/CollectSamErrorMetrics.java

```diff
@@ -232,12 +244,19 @@ protected int doWork() {
         final Collection<BaseErrorAggregation> aggregatorList = getAggregatorList();
         // Open up the input resources:
         try (
-                final SamReader sam = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
+                final SamReader sam = SamReaderFactory.makeDefault()
+                        .referenceSequence(REFERENCE_SEQUENCE)
+                        .setOption(SamReaderFactory.Option.INCLUDE_SOURCE_IN_RECORDS, true)
+                        .open(INPUT);
                 final ReferenceSequenceFileWalker referenceSequenceFileWalker = new ReferenceSequenceFileWalker(REFERENCE_SEQUENCE);
-                final PeekableIterator<VariantContext> vcfIterator = new PeekableIterator<>(
-                        VCF == null ? Collections.emptyIterator() : new VCFFileReader(VCF, true).iterator())
+                final VCFFileReader vcfFileReader = new VCFFileReader(VCF, true)
```

Comment:
why did you remove the option to not provide a VCF input? That would be a change in API...

Reply:
Oversight.  Fixed.

Reply:
@yfarjoun Actually as of 7/4/2018 VCF is not an optional argument.  (See line 147).

So this is not an API change.